### PR TITLE
Feat disable component when node is not built

### PR DESCRIPTION
**Pull request by EstherDarkish**
Disables all the interface components when the currentNode is disabled.

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
We have already `isNodeValid` function whichseems to be same than `isNodeDisabled` function ?
A node (model) is invalid and not disabled (component)

**Review by laurgarn — APPROVED** [review]

File: src/components/graph/util/model-functions.js [hunk 1]
```diff
@@ -30,9 +30,13 @@ export function getFirstNodeOfType(elements, nodeType, buildStatus) {
...
-export function isNodeValid(currentNode) {
-    const readOnlyNode = currentNode?.data?.readOnly;
-    const builtCurrentNode = currentNode?.data?.buildStatus === 'BUILT';
+export function isNodeReadOnly(node) {
+    if (node?.type === 'ROOT') return true;
+    return node?.data?.readOnly ? true : false;
+}
 
-    return !readOnlyNode && builtCurrentNode;
+export function isNodeDisabled(node) {
```
Comment (AbdelHedhili) [inline src/components/graph/util/model-functions.js:38]: as said by @SlimaneAmar rename to isNodeInvalid

File: src/components/study-container.js [hunk 2]
```diff
@@ -339,9 +345,9 @@ export function StudyContainer({ view, onChangeTab }) {
     }, [studyUuid, loadTree]);
 
     useEffect(() => {
-        if (!isNodeValid(currentNode) && currentNode?.type !== 'ROOT') return;
         loadNetwork(currentNode?.id === currentNodeIdRef.current);
```
Comment (AbdelHedhili) [inline src/components/study-container.js:348]: why are we not checking if the node is built anymore ?

File: src/components/network/equipment-table.js [hunk 3]
```diff
@@ -77,6 +77,7 @@ export const EquipmentTable = (props) => {
             )}
             {props.fetched &&
                 props.rows &&
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/network/equipment-table.js:79]: Yes ok. And also for `props.columns?.length > 0`.

File: src/components/app-top-bar.js [hunk 4]
```diff
@@ -242,7 +242,8 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
                 theme={themeLocal}
                 onEquipmentLabellingClick={handleChangeUseName}
                 equipmentLabelling={useNameLocal}
-                withElementsSearch={Boolean(studyUuid)}
+                withElementsSearch={true}
```
Comment (AbdelHedhili) [inline src/components/app-top-bar.js:245]: why this change ?

File: src/components/run-button-container.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ import makeStyles from '@mui/styles/makeStyles';
 import { addLoadflowNotif, addSANotif } from '../redux/actions';
 import { useDispatch, useSelector } from 'react-redux';
 import { useIntl } from 'react-intl';
-import { isNodeValid } from './graph/util/model-functions';
+import {} from './graph/util/model-functions';
```
Comment (AbdelHedhili) [inline src/components/run-button-container.js:19]: remove

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests:

. Sometimes, the 'LOG' component is totally empty (no disabled 'Log single node' button and no alert message 'Invalid Node')
. When an empty SLD is shown on an invalid node, and if we switch to 'SPREADSHEET' and return to 'MAP' view, the SLD is not shown anymore
. On an invalid node, the search bar is not disabled

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/components/run-button-container.js (hunk 5 above)
Comment (EstherDarkish) [reply src/components/run-button-container.js:19]: Nice catch

File: src/components/study-container.js (hunk 2 above)
Comment (AbdelHedhili) [reply src/components/study-container.js:348]: yes indeed no need then

**Comment by EstherDarkish** [comment]
> Tests:
> 
> . Sometimes, the 'LOG' component is totally empty (no disabled 'Log single node' button and no alert message 'Invalid Node') 

Fixed

> . When an empty SLD is shown on an invalid node, and if we switch to 'SPREADSHEET' and return to 'MAP' view, the SLD is not shown anymore 

This is not a trivial task. @JacquesBorsenberger can you help me with this ? When single-line-diagram-pane.js::284 is commented, we have the old Voltage Level error back.

> . On an invalid node, the search bar is not disabled

Fixed

File: src/components/app-top-bar.js (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/components/app-top-bar.js:245]: as the search function can now be disabled it prevent it from disappearing

**Review by FranckLecuyer — APPROVED** [review]

**Review by AbdelHedhili — APPROVED** [review]
